File: engine/src/main/java/com/bartlomiejpluta/base/engine/world/character/asset/CharacterSetAsset.java
package com.bartlomiejpluta.base.engine.world.character.asset;

import com.bartlomiejpluta.base.engine.common.asset.Asset;
import lombok.Getter;
import lombok.NonNull;

@Getter
public class CharacterSetAsset extends Asset {
   private final int rows;
   private final int columns;

   public CharacterSetAsset(@NonNull String uid, @NonNull String source, int rows, int columns) {
      super(uid, source);
      this.rows = rows;
      this.columns = columns;
   }
}

File: editor/src/main/kotlin/com/bartlomiejpluta/base/editor/characterset/asset/CharacterSetAsset.kt
package com.bartlomiejpluta.base.editor.characterset.asset

import com.bartlomiejpluta.base.editor.asset.model.GraphicAsset
import com.bartlomiejpluta.base.editor.project.model.Project

class CharacterSetAsset(project: Project, uid: String, source: String, name: String, rows: Int, columns: Int) :
   GraphicAsset(project.characterSetsDirectoryProperty, uid, source, name, rows, columns)